react/features/participants-pane/components/web/MeetingParticipantItem.js

```diff
@@ -235,7 +235,9 @@ function MeetingParticipantItem({
     }
 
     const buttonType = _isModerationSupported
-        ? _localModerator ? QUICK_ACTION_BUTTON.ASK_TO_UNMUTE : _quickActionButtonType
+        ? _localModerator && _audioMediaState !== MEDIA_STATE.UNMUTED
+            ? QUICK_ACTION_BUTTON.ASK_TO_UNMUTE
+            : _quickActionButtonType
```

Comment:
Why do we need to change the button type here ... when that is already calculated in _mapStateToProps?

Reply:
Why do we need here to calculate anything about audio levels here? 
Isn't it ok, to mix selectors/effects and connect method?
Are these useEffect and  useCallback calls ok, when you have 300 senders in the meeting?

Reply:
Please add tests for this case so we don't break it again.

Reply:
I'll update the function that calculates the quickActionButtonType.

> Why do we need to change the button type here ... when that is already calculated in _mapStateToProps?

Reply:
Because we want to display the green icon (the dominant speaker icon) not just for the dominant speaker but for any user that speaks (let's say multiple users speak at the same time, or when the dominant speaker talks there's someone else that makes noise) and that's why we need audio levels

> Why do we need to calculate anything about audio levels here, you can get the id of the current dominant speaker
> 
> https://github.com/jitsi/jitsi-meet/blob/40a485ec6c16b03ba2f00a5f8b472ae83761c9d6/react/features/base/participants/functions.js#L323
> 
> ?
> Is it ok to mix selectors/effects and connect method?
> Are these useEffect and useCallback calls ok, when you have 300 senders in the meeting?
> We had recently seen regression in participant pane when in 500 participant call ... can it be from these?

Reply:
I don't know. The regression for 500 participants could be from this, but this was done about 3 weeks ago

> Why do we need to calculate anything about audio levels here, you can get the id of the current dominant speaker
> 
> https://github.com/jitsi/jitsi-meet/blob/40a485ec6c16b03ba2f00a5f8b472ae83761c9d6/react/features/base/participants/functions.js#L323
> 
> ?
> Is it ok to mix selectors/effects and connect method?
> Are these useEffect and useCallback calls ok, when you have 300 senders in the meeting?
> We had recently seen regression in participant pane when in 500 participant call ... can it be from these?

Reply:
If we confirm that calculating audio levels is a problem for 500 participants meetings I could revert the changes. But I don't know how to run the 500 participants tests

Reply:
> Please add test for this case so we don't break it again.

The removeFromWhitelist test actually uses the Mute button so it's covered